k2/python/csrc/torch/ragged.cu

```diff
@@ -141,6 +141,72 @@ static void PybindRaggedImpl(py::module &m) {
   PybindRaggedTpl<int32_t>(m, "RaggedInt");
 }
 
+static void PybindRaggedShape(py::module &m) {
+  using PyClass = RaggedShape;
+  py::class_<PyClass> pyclass(m, "RaggedShape");
+  pyclass.def(py::init<const std::string &>(), py::arg("src"));
+  pyclass.def("dim0", &PyClass::Dim0);
+  pyclass.def("max_size", &PyClass::MaxSize, py::arg("axis"));
+  pyclass.def("num_axes", &PyClass::NumAxes);
+  pyclass.def("num_elements", &PyClass::NumElements);
+  pyclass.def("tot_size", &PyClass::TotSize, py::arg("axis"));
+
+  pyclass.def(
+      "to",
+      [](const PyClass &self, py::object device) -> PyClass {
+        std::string device_type = static_cast<py::str>(device.attr("type"));
+        K2_CHECK(device_type == "cpu" || device_type == "cuda")
+            << "Unsupported device type: " << device_type;
+
+        ContextPtr &context = self.Context();
+        if (device_type == "cpu") {
+          if (context->GetDeviceType() == kCpu) return self;
+          return self.To(GetCpuContext());
+        }
+
+        int32_t device_index = static_cast<py::int_>(device.attr("index"));
```

Comment:
Please have a look at 
https://github.com/k2-fsa/k2/blob/048da2a6f7fce1704f39a76c8dd5eef0b9b613fd/k2/python/csrc/torch/ragged.cu#L40-L43
and change `device_index` accordingly.

Reply:
fixed